dlio/stats.c

```diff
@@ -0,0 +1,1184 @@
+        }
+        fprintf(csv_file, "\", \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            fprintf(csv_file, "%cB/s", units[i]);
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        value = format_human_readable(eval_total_avg_read_rate);
+        fprintf(csv_file, "\"\neval avg raw read rate, %.3lf, %cB/s\n", value.value, value.unit);
+        fprintf(csv_file, "eval observed time per epoch, \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            fprintf(csv_file, "%.3lf", global_stats[i].observed_time.eval / 1000000.0);
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        fprintf(csv_file, "\", \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            fprintf(csv_file, "s");
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        fprintf(csv_file, "\"\neval observed time, %.3lf, s\n", eval_total_observed_time / 1000000.0);
+        fprintf(csv_file, "eval observed rate per epoch, \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            value    = format_human_readable(eval_avg_observed_rate_per_epoch[i]);
+            units[i] = value.unit;
+            fprintf(csv_file, "%.3lf", value.value);
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        fprintf(csv_file, "\", \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            fprintf(csv_file, "%cB/s", units[i]);
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        value = format_human_readable(eval_total_avg_observed_rate);
+        fprintf(csv_file, "\"\neval avg observed rate, %.3lf, %cB/s\n", value.value, value.unit);
+        fprintf(csv_file, "eval throughput samples per second per epoch, \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            fprintf(csv_file, "%.3lf", global_stats[i].throughput.eval);
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        fprintf(csv_file, "\", \"");
+        for (uint32_t i = 0; i < config.EPOCHS; i++) {
+            fprintf(csv_file, "samples/s");
+            if (i != config.EPOCHS - 1)
+                fprintf(csv_file, ", ");
+        }
+        fprintf(csv_file, "\"\neval throughput avg samples per second, %.3lf, samples/s\n",
+                eval_throughput_mean_samples_per_second);
+        fprintf(csv_file, "eval throughput stdev samples per second, %.3lf, samples/s\n",
+                eval_throughput_stdev_samples_per_second);
+        value = format_human_readable(eval_io_mean);
+        fprintf(csv_file, "eval io avg, %.3lf, %cB/s\n", value.value, value.unit);
+        value = format_human_readable(eval_io_stdev);
+        fprintf(csv_file, "eval io stdev, %.3lf, %cB/s\n", value.value, value.unit);
+    }
+
```

Comment:
Could you please paste/attach the sample output from a generated CSV? That way we can ensure it is using the same format as the other kernels.